runners/core-java/src/main/java/org/apache/beam/runners/core/InMemoryTimerInternals.java

```diff
@@ -167,20 +165,17 @@ public void deleteTimer(StateNamespace namespace, String timerId, TimeDomain tim
   @Deprecated
   @Override
   public void deleteTimer(StateNamespace namespace, String timerId, String timerFamilyId) {
-    TimerData existing = existingTimers.get(namespace, timerId + '+' + timerFamilyId);
-    if (existing != null) {
-      deleteTimer(existing);
+    TimerData removedTimer = existingTimers.remove(namespace, timerId + '+' + timerFamilyId);
+    if (removedTimer != null) {
+      timersForDomain(removedTimer.getDomain()).remove(removedTimer);
     }
   }
 
   /** @deprecated use {@link #deleteTimer(StateNamespace, String, TimeDomain)}. */
```

Comment:
It seems strange that the deprecated method is the one being used, and the recommended method is not implemented.

Reply:
yeah, actually that might have been a mistake when the timer family was added because the old way was to use `TimerData` and the new way is to use timerId/timerFamily.

Reply:
Maybe worth a JIRA/refactor? I'll leave it up to you since I am not as familiar with this code.